python/ray/train/v2/_internal/execution/controller/placement_group_cleaner.py

```diff
@@ -69,121 +60,111 @@ def _monitor_loop(self):
         """Monitor controller; remove PG when controller is gone.
 
         This runs continuously until controller dies or stop() is called.
+        Uses a queue to receive placement group updates.
         """
-        while not self._stopped and self._controller_actor_id:
+        while not self._stop_event.is_set():
+            # Check for new placement group updates from queue
+            try:
+                pg = self._pg_queue.get(timeout=self._check_interval_s)
+                self._curr_placement_group = pg
```

Comment:
nit: we could keep curr_placement_group as a local variable here, and just pass it into cleanup_placement_group